707.py

```diff
@@ -0,0 +1,91 @@
+class Node:
+
+    def __init__(self, val: int):
+        self.val = val
+        self.next = None
+
+class MyLinkedList:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.pre_head = Node(0)
+
+
+    def get(self, index: int) -> int:
+        """
+        Get the value of the index-th node in the linked list. If the index is invalid, return -1.
+        """
+        idx = 0
+        pointer = self.pre_head
+
+        while pointer.next and idx != index:
+            pointer = pointer.next
+            idx += 1
+
+        return pointer.next.val if pointer.next and idx == index else -1
+
+
+    def addAtHead(self, val: int) -> None:
+        """
+        Add a node of value val before the first element of the linked list. After the insertion, the new node will be the first node of the linked list.
+        """
+        head = Node(val)
+        head.next = self.pre_head.next
+        self.pre_head.next = head
+
+
+    def addAtTail(self, val: int) -> None:
```

Comment:
This `addAtTail()` operation can be done in `O(1)` time if you maintain a tail `Node` in your `MyLinkedList` class.